### ISSUE TITLE
Debug.fail hit  in bindThisPropertyAssignment

### ISSUE THREAD

**Issue by sheetalkamat** [Microsoft/TypeScript#22888]
**Code**

a.js file
```ts
this.a = 10;
```

tsc a.js --allowJS

**Expected behavior:**
no failure
**Actual behavior:**
```
Error: Debug Failure. SourceFile
    at bindThisPropertyAssignment (c:\TypeScript\built\local\tsc.js:23135:30)
    at bindWorker (c:\TypeScript\built\local\tsc.js:22848:29)
    at bind (c:\TypeScript\built\local\tsc.js:22739:13)
    at visitNode (c:\TypeScript\built\local\tsc.js:14173:24)
    at Object.forEachChild (c:\TypeScript\built\local\tsc.js:14412:24)
    at bindEachChild (c:\TypeScript\built\local\tsc.js:21543:16)
    at bindChildrenWorker (c:\TypeScript\built\local\tsc.js:21644:21)
    at bindChildren (c:\TypeScript\built\local\tsc.js:21513:17)
    at bind (c:\TypeScript\built\local\tsc.js:22750:21)
    at c:\TypeScript\built\local\tsc.js:21519:94
```

**Related Issues:**
Reported in https://github.com/Microsoft/TypeScript/issues/21140#issuecomment-375833399

**Comment by mhegazy** [comment]
@sandersn can you take a look for a fix for 2.8.2 please.

**Comment by sandersn** [comment]
The assert was added in #22643, which is not in 2.8, so the fix doesn't need to be in 2.8. I'll fix it for 2.9 instead.

**Comment by mhegazy** [comment]
thanks for the clarification.
